foundation/tests/org.jboss.tools.foundation.core.test/src/org/jboss/tools/foundation/core/properties/internal/SimpleHierarchicalVersionTest.java

```diff
@@ -25,7 +25,12 @@ public void testGetParentVersion() {
 		assertEquals("1.2.3.CR1-v20140211-1204",  new SimpleHierarchicalVersion("1.2.3.CR1-v20140211-1204-B52").getParentVersion().toString());
 		assertEquals("1.2.3.CR1-v20140211",  new SimpleHierarchicalVersion("1.2.3.CR1-v20140211-1204").getParentVersion().toString());
 		assertEquals("1.2.3.CR1",  new SimpleHierarchicalVersion("1.2.3.CR1-v20140211").getParentVersion().toString());
-		assertEquals("1.2.3", new SimpleHierarchicalVersion("1.2.3.CR1").getParentVersion().toString());
+		assertEquals("1.2.3.Alpha",  new SimpleHierarchicalVersion("1.2.3.Alpha1").getParentVersion().toString());
+		assertEquals("1.2.3.Beta",  new SimpleHierarchicalVersion("1.2.3.Beta20").getParentVersion().toString());
+		assertEquals("1.2.3.AM",  new SimpleHierarchicalVersion("1.2.3.AM3").getParentVersion().toString());
+		assertEquals("1.2.3.RC",  new SimpleHierarchicalVersion("1.2.3.RC4").getParentVersion().toString());
```

Comment:
Indeed there's no reason to use it for now. But the class could in theory be used to parse version for 3rd party plugins. Given it doesn't introduce any complexity, I added it.